init_unix.go

```diff
@@ -2,7 +2,7 @@
 
 package openssl
 
-// #cgo LDFLAGS: -ldl
+// #cgo LDFLAGS: -ldl -pthread
```

Comment:
I have a pedantic concern on hard-coding `-pthread` as some platforms seem to require `-lpthread` [instead](https://git.savannah.gnu.org/gitweb/?p=gnulib.git;a=blob;f=m4/threadlib.m4;h=6dee893ed8c9355f2e620094539b0a6729043a31;hb=HEAD#l634), though I guess it's out of scope of this module.

Reply:
I'm aware of this unfortunate conflict, but it is also truth that the Go toolchain uses `-pthread` when building cgo and nobody complained AFAIK. Let's keep it like this, we can always fix it if a bug is reported.